broker/src/main/java/io/zeebe/broker/system/partitions/impl/PartitionProcessingState.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.broker.system.partitions.impl;
+
+import io.atomix.raft.partition.RaftPartition;
+import io.zeebe.broker.Loggers;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import org.slf4j.Logger;
+
+public class PartitionProcessingState {
+
+  public static final Logger LOG = Loggers.SYSTEM_LOGGER;
+  private static final String PERSISTED_PAUSE_STATE_FILENAME = ".paused";
+  private boolean isProcessingPaused;
+  private final RaftPartition raftPartition;
+  private boolean diskSpaceAvailable;
+
+  public PartitionProcessingState(final RaftPartition raftPartition) {
+    this.raftPartition = raftPartition;
+    initProcessingStatus();
+  }
+
+  public boolean isDiskSpaceAvailable() {
+    return diskSpaceAvailable;
+  }
+
+  public void setDiskSpaceAvailable(final boolean diskSpaceAvailable) {
+    this.diskSpaceAvailable = diskSpaceAvailable;
+  }
+
+  public boolean isProcessingPaused() {
+    return isProcessingPaused;
+  }
+
+  @SuppressWarnings({"squid:S899"})
```

Comment:
what is this :eyes:

Reply:
Sonar complains that the returned value of `createNewFile` is not used. We are checking if the file exists in the next line.